src/Consolonia.Core/Helpers/Extensions.cs

```diff
@@ -12,5 +18,100 @@ public static void SubscribeAction<TValue>(
+        ///     emoji.
+        /// </summary>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        public static IList<string> GetGlyphs(this string text)
+        {
+            var console = AvaloniaLocator.Current.GetService<IConsole>();
+
+            var glyphs = new List<string>();
+            var emoji = new StringBuilder();
+            StringRuneEnumerator runes = text.EnumerateRunes();
+            var lastRune = new Rune();
+
+            while (runes.MoveNext())
+                if (console.SupportsComplexEmoji)
+                {
+                    if (lastRune.Value == Codepoints.ZWJ ||
+                        lastRune.Value == Codepoints.ORC ||
+                        Emoji.IsEmoji(runes.Current.ToString()))
+                    {
+                        emoji.Append(runes.Current);
+                    }
+                    else if (runes.Current.Value == Emoji.ZeroWidthJoiner ||
+                             runes.Current.Value == Emoji.ObjectReplacementCharacter ||
+                             runes.Current.Value == Codepoints.VariationSelectors.EmojiSymbol ||
+                             runes.Current.Value == Codepoints.VariationSelectors.TextSymbol)
+                    {
+                        emoji.Append(runes.Current);
+                    }
+                    else
+                    {
+                        if (emoji.Length > 0)
+                        {
+                            glyphs.Add(emoji.ToString());
+                            emoji.Clear();
+                        }
+
+                        glyphs.Add(runes.Current.ToString());
+                    }
+
+                    lastRune = runes.Current;
+                }
+                else
+                {
+                    if (runes.Current.Value != Emoji.ZeroWidthJoiner &&
+                        runes.Current.Value != Emoji.ObjectReplacementCharacter &&
+                        runes.Current.Value != Codepoints.VariationSelectors.EmojiSymbol &&
+                        runes.Current.Value != Codepoints.VariationSelectors.TextSymbol)
+                        glyphs.Add(runes.Current.ToString());
+                }
+
+            if (emoji.Length > 0) glyphs.Add(emoji.ToString());
+            return glyphs;
+        }
+
+        /// <summary>
+        ///     Measure text for actual width
+        /// </summary>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        public static ushort MeasureText(this string text)
+        {
+            var console = AvaloniaLocator.Current.GetService<IConsole>();
+
+            ushort width = 0;
+            ushort lastWidth = 0;
+            foreach (Rune rune in text.EnumerateRunes())
+            {
+                ushort runeWidth = (ushort)UnicodeCalculator.GetWidth(rune);
+                if (console.SupportsComplexEmoji &&
```

Comment:
Should it be saved to a method variable? Or we are assuming it will be optimized by the compiler?

Reply:
Oh enumerate runes gets called I've and returns an enumerator, foreach walks over the enumerator so this is efficient